sdk/aqueduct/utils/function_packaging.py

```diff
@@ -211,13 +218,10 @@ def _package_files_and_requirements(
             shutil.copy(REQUIREMENTS_FILE, packaged_requirements_path)
 
         else:
-            # No requirements have been provided, so we use `pip freeze` to infer.
-            logger().info(
-                "%s: No requirements.txt file detected, self-generating file by inferring package dependencies."
-                % func.__name__
+            # Do not infer package dependencies from environment.
+            raise FileNotFoundError(
+                "A valid requirements list or requirements.txt file must be provided"
```

Comment:
Let's give the user a little bit more guidance here. Can you add that the requirements.txt file should be in the same directory as the function definition? And that you can alternatively add the requirements directly in the @op/@metric/@check decorator.